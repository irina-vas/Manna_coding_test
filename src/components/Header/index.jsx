import React from 'react';
import { HeaderDesktop } from './HeaderDesktop';
import { HeaderMobile } from './HeaderMobile';

export const Header = () => {
  return (
    <>
     {window.innerWidth > 768 ? <HeaderDesktop /> : <HeaderMobile />}
    </>
  )
};
