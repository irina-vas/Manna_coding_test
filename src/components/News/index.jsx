/* eslint-disable react-hooks/exhaustive-deps */
import React, { useState, useEffect } from 'react';
import { dateFormatter } from '../../helpers';

export const News = ({ id, getNews, index }) => {
  const [news, setNews] = useState([]);
  useEffect(() => {
    getNews(id).then(data => setNews(data))
  }, []);

  return (
    <li className='news__item'>
      <a className="news__link" href={news.url} target="_blank" rel="noreferrer">
        <span className="news__number">{index + 4}.</span>
        <p className="news__title">{news.title}</p>
      </a>
      <div className="news__description">
        <div className="news__author">
          <p>by {news.by}</p>
          <p>{dateFormatter(news.time)}</p>
        </div>
        <div className="news__wrapper">
          <p className="news__score">score: {news.score}</p>
          <a className="news__comments" href={news.url} target="_blank" rel="noreferrer">
            comments: {news.descendants}
          </a>
      </div>
      </div>
    </li>
  )
}