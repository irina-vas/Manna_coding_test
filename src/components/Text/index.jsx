import { createElement } from 'react';
import cx from 'classnames';

export const Text = ({ children, tag, weight, size, color, className, ...rest }) => {
  const props = {
    className: cx('text',
      {[weight]: weight},
      {[size]: size},
      {[color]: color},
      className
    ),
    'data-testid': 'text',
    ...rest
  };
    return createElement(tag, props, children);
}