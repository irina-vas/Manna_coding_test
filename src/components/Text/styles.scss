.text.xlg {
  font-size: 65px;
  @include breakpoint(sm) {
    font-size: 48px;
  }
}
.text.lg {
  font-size: 22px;
}
.text.md {
  font-size: 14px;
}
.text.sm {
  font-size: 12px;
}
.text.w800 {
  font-weight: 800;
}
.text.w700 {
  font-weight: 700;
}
.text.w500 {
  font-weight: 500;
}
.text.w400 {
  font-weight: 400;
}
.text.black {
  color: #000;
}
.text.white {
  color: #fff;
}
.text.orange {
  color: #ff6600;
}
.text.opacity05 {
  color: rgba(256, 256, 256, 0.5);
  &:hover {
    color: rgba(256, 256, 256, 0.8)
  }
}
.text.opacity08 {
  color: rgba(256, 256, 256, 0.8);
}