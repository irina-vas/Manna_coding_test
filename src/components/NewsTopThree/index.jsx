/* eslint-disable react-hooks/exhaustive-deps */
import React, { useState, useEffect } from 'react';
import { dateFormatter } from '../../helpers';
import { Text } from '../Text';

export const NewsTopThree = ({ id, getNews, index }) => {
  const [news, setNews] = useState([]);

  useEffect(() => {
    getNews(id).then(data => setNews(data));
  }, []);

  return (
    <li className="newsTopThree__item">
      <a className="newsTopThree__link" href={news.url} target="_blank" rel="noreferrer">
        <Text tag="span" size="lg" weight="w800" color="orange" className="newsTopThree__number">
          {index + 1}
        </Text>
        <Text tag="p" color="white" weight="w400" size="md">{news.title}</Text>
      </a>
      <div className="newsTopThree__author">
        <Text tag="p" color="opacity05" weight="w400">by {news.by || 'unkhnown author'}</Text>
        <Text tag="p" color="opacity05" weight="w400">{dateFormatter(news.time)}</Text>
      </div>
      <div className="newsTopThree__wrapper">
        {<Text tag="p" color="opacity05" weight="w400">score: {news.score || 0}</Text>}
        <a className="newsTopThree__comments" href={news.url} target="_blank" rel="noreferrer">
          <Text tag="p" color="opacity05" weight="w400">comments: {news.descendants || 0}</Text>
        </a>
      </div>
    </li>
  )
}