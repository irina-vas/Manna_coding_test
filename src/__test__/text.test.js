/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-unnecessary-act */
import React from 'react';
import { createRoot } from 'react-dom/client';
import { Text } from '../components/Text';
import { act } from 'react-dom/test-utils';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('Text rendered without crashing', () => {
  const div = document.createElement('div');
  const root = createRoot(div);
  act(() => {
    root.render(<Text tag="span">Plain text</Text>)
  })
});

it('Text rendered properly', () => {
  const { getByTestId } = render(<Text tag="p">Plain text</Text>);
  expect(getByTestId('text')).toHaveTextContent('Plain text')
});

it('Matches snapshot', () => {
  const tree = renderer.create(<Text tag="p">Plain text</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});

