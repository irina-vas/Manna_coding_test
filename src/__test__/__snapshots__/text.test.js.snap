// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<p
  className="text"
  data-testid="text"
>
  Plain text
</p>
`;
